src/reference/asciidoc/gateway.adoc

```diff
@@ -319,6 +319,12 @@ Along with the `@MessagingGateway` annotation you can mark a service interface w
 
 Starting with version 6.0, an interface with the `@MessagingGateway` can also be marked with a `@Primary` annotation for respective configuration logic as its possible with any Spring `@Component` definition.
 
+The `@MessagingGateway.name()` attribute specifies a name for the target bean registered.
+If it is not provided, the  `@IntegrationComponentScan.nameGenerator()` (since version `6.0`) is consulted.
+And if that is not provided as well, an `AnnotationConfigUtils.CONFIGURATION_BEAN_NAME_GENERATOR` bean, provided for the application context, is used.
+Otherwise, the bean name generation falls back to the `AnnotationBeanNameGenerator` which results with an uncapitalized simple class name.
+In other words, the current naming strategy for gateway proxy beans is aligned with standard Spring Framework `@ComponentScan` approach.
+
```

Comment:
```suggestion
The `@MessagingGateway.name()` attribute specifies a name for the target bean that will be registered.
If it is not provided, the  `@IntegrationComponentScan.nameGenerator()` (since version `6.0`) is consulted.
And, if that is not provided, an `AnnotationConfigUtils.CONFIGURATION_BEAN_NAME_GENERATOR` bean, if present in the application context, is used.
Otherwise, the bean name generation falls back to the `AnnotationBeanNameGenerator` which results with an uncapitalized simple class name.
In other words, the current naming strategy for gateway proxy beans is aligned with standard Spring Framework `@ComponentScan` approach.

```